pkg/cmd/api/api.go

```diff
@@ -0,0 +1,142 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/cli/v3/pkg/cmd/validation"
+	"github.com/spf13/cobra"
+)
+
+var (
+	method    string
+	headers   []string
+	data      string
+	analytics bool
+	queryFile string
+)
+
+func NewCmdAPI(f *factory.Factory) *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "api <endpoint>",
+		Short: "Interact with the Buildkite API",
+		Long:  "Interact with either the REST or GraphQL Buildkite APIs",
+		Args:  cobra.MaximumNArgs(1),
+		Example: heredoc.Doc(`
+      # To get a build
+      $ bk api /pipelines/example-pipeline/builds/420
+
+      # To create a pipeline
+      $ bk api --method POST /pipelines --data '
```

Comment:
It would even be cool if we automatically used POST when data is included since its not valid for GET anyway. And then people can specify PUT/PATCH when necessary to change that. A thing for another PR though

Reply:
Hmm, the problem is you also pass `data` to a PUT, so setting it to POST as default might cause confusion